### ISSUE TITLE
[RFC] Integrate tf-keras example and keras example

### ISSUE THREAD

**Issue by crcrpar** [optuna/optuna#1479]
Now that Keras doesn't support backends other than TensorFlow as announced in https://github.com/keras-team/keras/releases/tag/2.4.0, we can integrate them into one without loss of generality.

However, removing an example is a bit too drastic. So, I would love to discuss whether we integrate the two examples or leave them as they are.

**Comment by hvy** [comment]
Thanks for raising this issue. On a related note, we should probably support `keras==2.4.0` first (see `keras<2.4.0` https://github.com/optuna/optuna/blob/master/setup.py).

**Comment by crcrpar** [comment]
One thing in my mind is that now that Optuna avoids Keras 2.4 due to the lack of compatibility between Keras and TensorFlow (when I tackled the CI failure about two or three weeks ago). So, we have to defer the decision about when to start cleaning up Keras and TF-Keras examples.

**Comment by crcrpar** [comment]
I'm going to check the compatibility of Keras and TensorFlow upon TensorFlow 2.3 release.

**Comment by crcrpar** [comment]
Ah, it looks like we can move forward now as the limitation has gone: https://github.com/optuna/optuna/pull/1563.

**Comment by hvy** [comment]
@crcrpar Do you have any suggestions on how to proceed with this issue?

I suggest we take into account the two integration modules as well, the one for `keras` and the other one for `tf.keras`. Given the release note of `keras` referenced from above, namely that there is only "one Keras: `tf.keras`, it seems like we should proceed with deprecating the former, along with removing the example. I'm not an expert with the TF ecosystem, so curious what your take is.

**Comment by crcrpar** [comment]
Personally I am trending toward integrate Keras integration and example into tf.Keras integration and example. Hence, TODOs are as follows:
1. Deprecate Keras integration and example in the shoes of removing Keras ones in the next major release
2. File an issue for the above tasks as contribution-welcome

**Comment by crcrpar** [comment]
Now committers agreed on deprecating Keras ones. So closing.
